### PR TITLE
updates to cloud trials

### PR DESCRIPTION
**Pull request by amenne**
This PR does the following:

1. Updates to cloud trial language, removing previous restrictions and reference to 30 day trials (account team will align on timeline)
2. Updates references to features compatible with cloud
3. Removes section on Cody and instead directs user to Cody pages

### CODE REVIEW

File: docs/cloud/index.mdx [hunk 1]
```diff
@@ -40,11 +31,13 @@ As part of this service you will receive a number of benefits from our team, inc
 
 ### Access to all Sourcegraph features
 
-All Sourcegraph features are avilable on Sourcegraph Cloud instances out-of-the-box, including but not limited to:
+All of Sourcegraph's features are available on Sourcegraph Cloud instances out-of-the-box, including but not limited to:
 
+- [Cody](../cody/index.mdx)
```
Comment (MaedahBatool) [inline docs/cloud/index.mdx:36]:
```suggestion
- [Cody](/cody)
```
Link path fix

File: docs/cloud/index.mdx [hunk 2]
```diff
@@ -40,11 +31,13 @@ As part of this service you will receive a number of benefits from our team, inc
...
 - [Server-side Batch Changes](/batch_changes/explanations/server_side)
 - [Precise code navigation powered by auto-indexing](/code_navigation/explanations/auto_indexing)
 - [Code Monitoring](/code_monitoring/) (including [email delivery](#managed-smtp) of notifications)
+- [Code Insights](../code_insights/index.mdx)
```
Comment (MaedahBatool) [inline docs/cloud/index.mdx:40]:
```suggestion
- [Code Insights](/code_insights/)
```
Link path fix.

**Review by MaedahBatool — CHANGES_REQUESTED** [review]
Thanks for contributing. Left some small feedback.

**Comment by amenne** [comment]
Thanks @MaedahBatool committed suggestions on the links. Good to merge?

**Comment by MaedahBatool** [comment]
@amenne yes, we are good to go. I am merging it. :)

**Review by MaedahBatool — APPROVED** [review]

**Pull request merged**